src/gate_detect/src/GateDetectorNode.cpp

```diff
@@ -72,68 +70,86 @@ const sensor_msgs::ImageConstPtr& msg) {
+                                   gateCoordinates.distanceLeftPole);
+    }
+    if (gate.rightDetected) {
+        gateCoordinates.detectedRightPole = true;
+        int rightWidth                    = gate.rightPole.getVertWidth();
+        gateCoordinates.distanceRightPole =
+        interpolator_.getVertDistance(rightWidth);
+        gateCoordinates.angleRightPole =
+        interpolator_.getVertAngle(gate.rightPole.getVertMid(),
+                                   rightWidth,
+                                   image.cols,
+                                   gateCoordinates.distanceRightPole);
+    }
+    if (gate.topDetected) {
+        gateCoordinates.detectedTopPole = true;
+        int topWidth                    = gate.topPole.getHorWidth();
+        gateCoordinates.distanceTopPole =
+        interpolator_.getHorDistance(topWidth);
+        gateCoordinates.angleTopPole =
+        interpolator_.getHorAngle(gate.topPole.getHorMid(),
+                                  topWidth,
+                                  image.rows,
+                                  gateCoordinates.distanceTopPole);
+    }
     publishGateDetectMsg(gateCoordinates);
-
-    if (displayDetectedGate_) publishGateImage(gateCoordinates);
 }
 
 void GateDetectorNode::publishGateDetectMsg(GateCoordinates gateCoordinates) {
     gate_detect::GateDetectMsg msg;
 
-    if (gateCoordinates.detectedLeftPole == 0.0) {
-        msg.detectedLeftPole = 0;
-        msg.angleLeftPole    = 0;
-        msg.distanceLeftPole = 0;
-    } else {
-        msg.detectedLeftPole = 1;
-        msg.angleLeftPole    = gateCoordinates.angleLeftPole;
-        msg.distanceLeftPole = gateCoordinates.distanceLeftPole;
-    }
-    if (gateCoordinates.angleRightPole == 0.0) {
-        msg.detectedRightPole = 0;
-        msg.angleRightPole    = 0;
-        msg.distanceRightPole = 0;
-
-    } else {
-        msg.detectedRightPole = 1;
-        msg.angleRightPole    = gateCoordinates.angleRightPole;
-        msg.distanceRightPole = gateCoordinates.distanceRightPole;
-    }
-    if (gateCoordinates.detectedTopPole == 0.0) {
-        msg.detectedTopPole = 0;
-        msg.angleTopPole    = 0;
-        msg.distanceTopPole = 0;
-
-    } else {
-        msg.detectedTopPole = 1;
-        msg.angleTopPole    = gateCoordinates.angleTopPole;
-        msg.distanceTopPole = gateCoordinates.distanceTopPole;
-    }
+    msg.detectedLeftPole = gateCoordinates.detectedLeftPole;
+    msg.angleLeftPole    = gateCoordinates.angleLeftPole;
+    msg.distanceLeftPole = gateCoordinates.distanceLeftPole;
+
+    msg.detectedRightPole = gateCoordinates.detectedRightPole;
+    msg.angleRightPole    = gateCoordinates.angleRightPole;
+    msg.distanceRightPole = gateCoordinates.distanceRightPole;
+
+    msg.detectedTopPole = gateCoordinates.detectedTopPole;
+    msg.angleTopPole    = gateCoordinates.angleTopPole;
+    msg.distanceTopPole = gateCoordinates.distanceTopPole;
 
     publisher1_.publish(msg);
```

Comment:
Perhaps `gate_msg_publisher` might be a bit more descriptive?

Reply:
Done